go/store/val/custom_comparator.go

```diff
@@ -0,0 +1,94 @@
+// Copyright 2024 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package val
+
+// CustomTupleComparator is a comparator that properly handles custom types.
+type CustomTupleComparator struct {
```

Comment:
i agree, why's the indirection necessary? `SerializedCompare` also seems like a thing that you'd want hidden in the interface, rather than duplicated for a subset of hardcoded types